### DISCUSSION TITLE
Can't understand how work with layers

### DISCUSSION THREAD

**Discussion by boriborm** [ImageMagick/ImageMagick#3319] [Help]
Hi,
I run this steps:
1. convert in.jpg  \( +clone -blur 0x5 \) -compose Divide_Src -composite -normalize -blur 0x1 t1.jpg"
2. convert in.jpg -morphology Convolve DoG:5,30,0 -negate -normalize -blur 0x2 -morphology Erode Disk:1 -level 70%,91%,0.1 -threshold 50% -negate t2.jpg"
3. convert t1.jpg t2.jpg -compose Minus_Dst -composite out.jpg

How can i write IM script without using t1.jpg and t2.jpg?

**Comment by snibgo** [comment]
What version of IM? On what platform (bash, Windows BAT, etc)?

**Reply by boriborm** [reply]
IM version 6.8.9, Ubuntu, bash.

**Comment by snibgo** [comment]
For bash, I'm surprised the parentheses didn't need to be escaped.

Something like this (untested):
```
convert \
  in.jpg +write mpr:INP \
  \( +clone -blur 0x5 \) -compose Divide_Src -composite -normalize -blur 0x1 \
  \( mpr:INP \
     -morphology Convolve DoG:5,30,0 -negate -normalize -blur 0x2 \
     -morphology Erode Disk:1 -level 70%,91%,0.1 -threshold 50% -negate \
  \) \
  -compose Minus_Dst -composite \
  out.jpg
```

**Reply by boriborm** [reply]
Thank you! It works fine...

**Reply by goyalyashpal** [reply]
mark it as answer 😊
